pkg/aws/aws.go

```diff
@@ -286,27 +286,99 @@ func (c *Client) ClusterSharedIamOperation(operation string) error {
 
 	switch operation {
 	case "add":
-		role, err := PutIAMRole(roleName, "shared cluster role", roleDocument, iamClient)
+		role, err := PutIAMRole(roleName, "shared cluster role", roleDocument, c.IAMClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to create shared cluster role")
 		}
-		log.Infof("BDD >> created shared iam role: %s", awssdk.StringValue(role.Arn))
+		log.Infof("BDD >> created shared iam role: %s", aws.StringValue(role.Arn))
 
-		policy, err := PutManagedPolicy(roleName, clusterSharedPolicy, "shared cluster policy", policyDocument, iamClient)
+		policy, err := PutManagedPolicy(roleName, clusterSharedPolicy, "shared cluster policy", policyDocument, c.IAMClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to create shared cluster managed policy")
 		}
-		log.Infof("BDD >> created shared iam policy: %s", awssdk.StringValue(policy.Arn))
+		log.Infof("BDD >> created shared iam policy: %s", aws.StringValue(policy.Arn))
 	case "remove":
-		err := DeleteManagedPolicy(clusterSharedPolicy, iamClient)
+		err := DeleteManagedPolicy(clusterSharedPolicy, c.IAMClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to delete shared cluster role")
 		}
 
-		err = DeleteIAMRole(roleName, iamClient)
+		err = DeleteIAMRole(roleName, c.IAMClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to delete shared cluster managed policy")
 		}
 	}
 	return nil
 }
+
+func (c *Client) GetEksVpc() (string, error) {
+	clusterName, err := util.GetClusterName()
+	if err != nil {
+		return "", err
+	}
+	input := &eks.DescribeClusterInput{
+		Name: aws.String(clusterName),
+	}
+	result, err := c.EKSClient.DescribeCluster(input)
+	if err != nil {
+		return "", err
+	}
+	return aws.StringValue(result.Cluster.ResourcesVpcConfig.VpcId), nil
+}
+
+func (c *Client) DnsNameInHostedZoneID(dnsName string, hostedZoneID string) error {
+	time.Sleep(120 * time.Second)
```

Comment:
Should this be configurable? Maybe we could also log something to inform user what delay is for?

Reply:
I dont know if that is actually needed. I will implement a generic step to wait. If we need to, we can use that before this one in the `.feature` file.